### PR TITLE
Use full urls to allow easy submodule init after cloning from fork

### PR DESCRIPTION
**Pull request by tuxlifan**

### CODE REVIEW

**Comment by nthdimtech** [comment]
I think this choice is problematic. Awhile back I accepted a pull request to make them relative...

This change makes a fork of *only* the client require no changes to the submodule paths at the cost of making a fork of the larger subsets of the project require path changes. There are many proposed firmware level features that have client components so forks made for that purposes of implementing those would create merge problems. This change also makes mirroring the repositories similarly problematic. I have considered mirroring the sources on nthdimtech.com for instance.

I think I will just make the recommendation that people either fork everything or alter the paths in private uncommitted changes to the submodule paths.

**Comment by tuxlifan** [comment]
Yes, I confess the thought that you (or someone) might rely on or use "../" leading to local clones had briefly crossed my mind.

I've instead now created a script that does the url mangling, calls git submodule update --init and then places the original file back to make git think nothing changed for .gitmodules.
This comes in handy in case someone wants to fork only signet-desktop-client to hack on it without having to care about the submodules.

The only caveat is that if one does a manual git submodule update --init first and it fails due to "../" paths not resolving those have to be manually deleted from .git/config before running contrib/git-init-upstream-submodules.sh

So to recap:
1. cloning from https://github.com/nthdimtech/signet-desktop-client:
  run `git submodule update --init`

2. cloning from a fork/location which has all repos:
 run `git submodule update --init`

3. cloning from a fork/location which has only signet-desktop-client:
 run `./contrib/git-init-upstream-submodules.sh`

**Comment by nthdimtech** [comment]
I like this approach but I see one problem. The script calls "git submodule update" but I think it should also be calling "git submodule sync" so that if by chance the submodules are already initialized they can still pick up the new locations

**Comment by tuxlifan** [comment]
Done.
